lib/bloc/lnurl/lnurl_bloc.dart

```diff
@@ -69,8 +71,12 @@ class LNUrlBloc with AsyncActionsHandler {
         _lnUrlStreamController.add(fetchLNUrlState.completed);
         if (response.runtimeType == LNUrlResponse) {
           if (response.hasWithdraw()) {
-            _lnUrlStreamController
-                .add(WithdrawFetchResponse(response.withdraw));
+            final withdrawResponse = WithdrawFetchResponse(response.withdraw);
+            if (withdrawFetchResponseInterceptor != null) {
+              withdrawFetchResponseInterceptor.intercept(withdrawResponse);
```

Comment:
A lot better but the issue here is that still the logic is delegated to the UI instead of being executed here.
If the UI would have registered only the invoice (withdrawalInvoice) here (instead of the interceptor) the bloc could handle this automatically and complete the lnurl flow by invoking _withdraw function directly.
Wouldn't it be cleaner in your opinion?

Reply:
That makes sense, let me see if that works 👀

Reply:
I did move the logic to the bloc, I think it is ready for a new review @roeierez